lib/portlayer/storage/image_cache.go

```diff
@@ -366,6 +370,72 @@ func (c *NameLookupCache) DeleteImage(op trace.Operation, image *Image) error {
 	return nil
 }
 
+// DeleteBranch deletes a branch of images, starting from nodeID, up to the
+// first node with degree greater than 1.  keepNodes is the array of images to
+// keep (and their branches).
+func (c *NameLookupCache) DeleteBranch(op trace.Operation, image *Image, keepNodes []*url.URL) ([]*Image, error) {
+	op.Infof("DeleteBranch: deleting branch starting at %s", image.Self())
+
+	var (
+		deletedImages []*Image
+		err           error
+	)
+
+	// map of images to keep
+	keep := make(map[url.URL]int)
+	for _, elem := range keepNodes {
+		keep[*elem] = 0
+	}
+
+	// Check if the error is actually an error.  If we deleted something,
+	checkErr := func(err error, deleted []*Image) ([]*Image, error) {
```

Comment:
I'd rather leave it where it is in case someone is tempted to use it for unrelated things.  I refuse to use a `goto` in `go` (and that's what I'd do if this was `C`), so this is close enough.